File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/analytics/FirebaseAnalyticsAppLifecycleCallback.java
package com.jdroid.android.firebase.analytics;

import android.content.Context;

import com.jdroid.android.application.AbstractApplication;
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback;

public class FirebaseAnalyticsAppLifecycleCallback extends ApplicationLifecycleCallback {
	@Override
	public void onProviderInit(Context context) {
		AbstractApplication.get().addCoreAnalyticsTracker(new FirebaseCoreAnalyticsTracker());
	}
	
	@Override
	public Boolean isEnabled() {
		return FirebaseAnalyticsFactory.getFirebaseAnalyticsHelper().isFirebaseAnalyticsEnabled();
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/analytics/FirebaseAnalyticsAppLifecycleCallback.kt
package com.jdroid.android.firebase.analytics

import android.content.Context
import com.jdroid.android.application.AbstractApplication
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback

class FirebaseAnalyticsAppLifecycleCallback : ApplicationLifecycleCallback() {

    override fun onProviderInit(context: Context) {
        AbstractApplication.get().addCoreAnalyticsTracker(FirebaseCoreAnalyticsTracker())
    }

    override fun isEnabled(): Boolean {
        return FirebaseAnalyticsHelper.isFirebaseAnalyticsEnabled()
    }
}
